brillig_bytecode/src/lib.rs

```diff
@@ -0,0 +1,847 @@
+                self.increment_program_counter()
+            }
+            Opcode::Trap => self.fail("explicit trap hit in brillig"),
+            Opcode::Stop => self.finish(),
+            Opcode::Load { destination: destination_register, source_pointer } => {
+                // Convert our source_pointer to a usize
+                let source = self.registers.get(*source_pointer);
+                // Use our usize source index to lookup the value in memory
+                let value = &self.memory[source.to_usize()];
+                self.registers.set(*destination_register, *value);
+                self.increment_program_counter()
+            }
+            Opcode::Store { destination_pointer, source: source_register } => {
+                // Convert our destination_pointer to a usize
+                let destination = self.registers.get(*destination_pointer);
+                // Use our usize destination index to set the value in memory
+                self.memory[destination.to_usize()] = self.registers.get(*source_register);
+                self.increment_program_counter()
+            }
+            Opcode::Call { location } => {
+                // Push a return location
+                self.call_stack.push(Value::from(self.program_counter + 1));
+                self.set_program_counter(*location)
+            }
+            Opcode::Const { destination, value } => {
+                self.registers.set(*destination, *value);
+                self.increment_program_counter()
+            }
+        }
+    }
+
+    pub fn program_counter(self) -> usize {
+        self.program_counter
+    }
+
+    /// Increments the program counter by 1.
+    fn increment_program_counter(&mut self) -> VMStatus {
+        self.set_program_counter(self.program_counter + 1)
+    }
+
+    /// Increments the program counter by `value`.
+    /// If the program counter no longer points to an opcode
+    /// in the bytecode, then the VMStatus reports halted.
+    fn set_program_counter(&mut self, value: usize) -> VMStatus {
+        assert!(self.program_counter < self.bytecode.len());
+        self.program_counter = value;
+        if self.program_counter >= self.bytecode.len() {
+            self.status = VMStatus::Finished;
+        }
+        self.status.clone()
+    }
+
+    /// Process a binary operation.
+    /// This method will not modify the program counter.
+    fn process_binary_int_op(
+        &mut self,
+        op: BinaryIntOp,
+        bit_size: u32,
+        lhs: RegisterIndex,
+        rhs: RegisterIndex,
+        result: RegisterIndex,
+    ) {
+        let lhs_value = self.registers.get(lhs);
+        let rhs_value = self.registers.get(rhs);
+
+        let result_value = op.evaluate_int(lhs_value.to_u128(), rhs_value.to_u128(), bit_size);
+
+        self.registers.set(result, result_value.into());
+    }
+
+    fn resolve_foreign_call_input(&self, input: RegisterValueOrArray) -> Vec<Value> {
```

Comment:
This method in isolation does not seem to have anything to do with `foreign_call_inputs` -- Might need a rename

Reply:
Created issue #306